usr/lib/pkcs11/ep11_stdll/ep11_specific.c

```diff
@@ -3504,9 +3504,49 @@ CK_RV ep11tok_derive_key(STDLL_TokData_t * tokdata, SESSION * session,
     unsigned char *ep11_pin_blob = NULL;
     CK_ULONG ep11_pin_blob_len = 0;
     ep11_session_t *ep11_session = (ep11_session_t *) session->private_data;
+    CK_ECDH1_DERIVE_PARAMS *ecdh1_parms;
+    CK_MECHANISM ecdh1_mech;
 
     memset(newblob, 0, sizeof(newblob));
 
+    /*
+     * EP11 supports CKM_ECDH1_DERIVE slightly different than specified in
+     * PKCS#11. It expects the public data directly as mechanism param,
+     * not via CK_ECDH1_DERIVE_PARAMS. It also does not support KDFs and shared
+     * data.
+     *
+     * ATTENTION: Once the EP11 library supports CKM_ECDH1_DERIVE in PKCS#11
+     * way, the following code needs to be removed. Possibly a EP11 library
+     * version check needs to be added to conditionally remove the following
+     * code.
+     */
```

Comment:
id just mention the code has to be adapted in this case. if it has to be removed or be part of a if-version>-else block will be determined when its time. could you mark the begin and end of the code block to which this comment refers or just say "the following if block". When you cite pkcs11 spec could you add the version to which you refer (2.2.0).